packages/flutter/lib/src/gestures/mouse_tracking.dart

```diff
@@ -133,18 +145,19 @@ class MouseTracker extends ChangeNotifier {
     _trackedAnnotations.remove(annotation);
   }
 
-  bool _postFrameCheckScheduled = false;
+  bool _scheduledPostFramePositionCheck = false;
+  // Schedules a position check at the end of this frame for those annotations
+  // that have been added.
   void _scheduleMousePositionCheck() {
     // If we're not tracking anything, then there is no point in registering a
     // frame callback or scheduling a frame. By definition there are no active
     // annotations that need exiting, either.
-    if (_trackedAnnotations.isNotEmpty && !_postFrameCheckScheduled) {
-      _postFrameCheckScheduled = true;
-      SchedulerBinding.instance.addPostFrameCallback((Duration _) {
-        _postFrameCheckScheduled = false;
-        collectMousePositions();
+    if (_trackedAnnotations.isNotEmpty && !_scheduledPostFramePositionCheck) {
+      _scheduledPostFramePositionCheck = true;
+      SchedulerBinding.instance.addPostFrameCallback((Duration duration) {
+        sendMouseNotifications(_lastMouseEvent.keys.toSet());
+        _scheduledPostFramePositionCheck = false;
       });
-      SchedulerBinding.instance.scheduleFrame();
```

Comment:
Is this removal safe? Does this properly schedule a frame when an annotation is attached?

Reply:
This is actually the point of the PR.  We have to remove this because it schedules frames every time the mouse moves.  We don't need to do this, because when adding annotations we're already rendering a frame, so it's sufficient to just schedule a post frame callback. For other cases, it is now synchronous, so no new frame is needed.

Reply:
`_scheduleMousePositionCheck` is only called when an annotation is added now so we don't have to remove it I suppose. But it's ok too if adding annotations already means rendering a frame.